### PR TITLE
Feat/add option to show references in qf

### PR DESCRIPTION
**Pull request by andlrc**
Add the option "useQuickfixForLocations" to use the quickfix list for locations comming from ":LspShowReferences", ":LspGotoDefinition", etc.

### CODE REVIEW

File: doc/lsp.txt [hunk 1]
```diff
@@ -272,8 +272,29 @@ server and to set the language server options.  For example: >
...
+<
+The LspOptionsGet() function returns a |Dict| of all the LSP plugin options,
+To get a particular option value you can use the following: >
+
+	echo LspOptionsGet()['autoHighlightDiags']
+<
+
+All options are listed below:
+
+							*lsp-options-func*
```
Comment (andlrc) [inline doc/lsp.txt:297]: This split of options might not make sense? I think the list of options is growing quite large though.

File: autoload/lsp/options.vim [hunk 2]
```diff
@@ -72,6 +72,8 @@ export var lspOptions: dict<any> = {
   diagSignInfoTexthl: 'Pmenu',
   diagSignHintText: 'H>',
   diagSignHintTexthl: 'Question',
+  # ShowReferences in a quickfix list instead of a location list`
+  referencesInQf: false
```
Comment (andlrc) [inline autoload/lsp/options.vim:76]: This option might be named poorly, as it is also used when showing multiple implemention/definitions/etc in a list.

Comment (yegappan) [reply autoload/lsp/options.vim:76]: What about using the name "useQuickfix"?

File: doc/lsp.txt (hunk 1 above)
Comment (yegappan) [reply doc/lsp.txt:297]: I am thinking of removing the highlight group related options.  The user can override the pre-defined highlight groups by defining them before sourcing the plugin.

Comment (andlrc) [reply doc/lsp.txt:297]: Yes, that makes perfect sense.

Comment (andlrc) [reply doc/lsp.txt:297]: I'll remove the commit that reorders the docs then.

File: autoload/lsp/options.vim (hunk 2 above)
Comment (andlrc) [reply autoload/lsp/options.vim:76]: > What about using the name "useQuickfix"?

We have the option "autoPopulateDiags" which populates the location list, and I want to keep it like that, as then with this new option it's possible to use both.

What about "useQuickfixForLocations"

**Pull request merged**